google/cloud/mocks/current_options.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_MOCKS_CURRENT_OPTIONS_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_MOCKS_CURRENT_OPTIONS_H
+
+#include "google/cloud/options.h"
+
+namespace google {
+namespace cloud {
+namespace mocks {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * Retrieve options used in a client call.
+ *
+ * This would be used to verify configuration options from within a
+ * MockConnection. It provides a way for applications to test the difference
+ * between `client.Foo(request, options)` and `client.Foo(request)`.
+ *
+ * @code
+ * TEST(Foo, CallOptions) {
+ *   auto mock = std::make_shared<MockConnection>();
+ *   EXPECT_CALL(*mock, Foo).WillOnce([] {
+ *         auto const& options = google::cloud::mocks::CurrentOptions();
```

Comment:
If `CurrentOptions()` is going to return by value, then I wouldn't suggest capturing that value by `const&`.

Reply:
changed to `Options const& mocks::CurrentOptions()`